### PR TITLE
Adds Chemical Explosives

### PR DESCRIPTION
**Pull request by nullbear**
:cl: nullbear
rscadd: Added several craftable explosive chemical weapons.
rscadd: Made it possible to craft empty bomb assemblies using plasteel. Note: they will need a payload, and to have their wires set, before it can be detonated.
tweak: Large Grenades are 10% larger.
tweak: Grenades heat up their contents when they explode.
tweak: Non-grenade reagents are applied per turf affected, rather than per atom.
/:cl:

### CODE REVIEW

**Comment by Cheridan** [comment]
That's not how we do changelogs these days

**Comment by nullbear** [comment]
what do you mean ? your wiki told me to do it like that.

**Comment by Cheridan** [comment]
It's outdated then

**Comment by nullbear** [comment]
what do i do then? The example changelog says to do this too.

**Comment by ike709** [comment]
see literally any other PR with a changelog

**Comment by nullbear** [comment]
Better?

File: code/game/objects/items/weapons/grenades/chem_grenade.dm [hunk 1]
```diff
@@ -272,6 +224,76 @@
...
+	ignition_temp = -100
+
+/obj/item/weapon/grenade/chem_grenade/pyro // Intended for pyrotechnical mixes. Produces a small fire upon detonation, igniting potentially flammable mixtures.
+	name = "pyro grenade"
+	desc = "A custom made pyrotechnical grenade. It heats up and ignites its contents upon detonation."
+	icon_state = "pyrog"
+	affected_area = 3
+	ignition_temp = 500 // This is enough to expose a hotspot.
+
+/obj/item/weapon/grenade/chem_grenade/adv_release // Intended for weaker, but longer lasting effects. Could have some interesting uses.
```
Comment (Iamgoofball) [inline code/game/objects/items/weapons/grenades/chem_grenade.dm]: HNNNNGH

:+1:

**Comment by Bawhoppen** [comment]
don't bundle a bunch of mostly unrelated changes together, 
make a seperate PR for the stuff like buffing grenades, and the setting off bombs part
also don't make bombs be emaggable don't you dare

**Comment by nullbear** [comment]
Well i made the bombs emaggable cause the minimum timer was changed from 60 to 120. So emaggable bombs just removes the nerf for antags.

As for the buffing grenades and stuff. That's all part of the chem_splash proc that they're all reliant on. Beforehand, the splash would do something like "units / 5-100,000+" where you'd end up with like 0.00001 units being applied to everything splashed. The 'buff' changes it to apply the units per turf, rather than per object, so it's much more consistent, and will typically have an actual effect instead of not doing anything. This was tested by filling a grenade with 600 units of acid in the bar. Using current grenades, nothing at all happens.

The rest of the unrelated stuff could be put to a different pr, sure. I figured they were related, though.

**Comment by nullbear** [comment]
@gunhog how is this looking?

**Comment by Iamgoofball** [comment]
ignore bawhoppen he's dumb

File: code/game/machinery/syndicatebomb.dm [hunk 2]
```diff
@@ -292,6 +326,127 @@
...
+
+		for(var/obj/item/weapon/reagent_containers/glass/B in G)
+			if(beakers.len < max_beakers)
+				beakers += B
+				B.loc = src
+			else
+				B.loc = get_turf(src)
+
+		qdel(G)
+	return
```
Comment (RemieRichards) [inline code/game/machinery/syndicatebomb.dm]: trailing return, remove it please.

Comment (nullbear) [inline code/game/machinery/syndicatebomb.dm]: I don't know what that means.

**Comment by Supermichael777** [comment]
do note that that timers is extremely inaccurate and 60 seconds typically translates into 2 minutes or more. also you could make the minimum time a var and have a sub-type with a higher number for crafted ones. emags do way to much already

**Comment by nullbear** [comment]
Yeah, that's what i ended up going with, @Supermichael777

File: code/game/machinery/syndicatebomb.dm (hunk 2 above)
Comment (GunHog) [inline code/game/machinery/syndicatebomb.dm]: You do not have to include the "return" command at the end of a proc, as it will do so automatically.

**Pull request merged**